include/pybind11/numpy.h

```diff
@@ -561,35 +595,36 @@ class array : public buffer {
     }
 };
 
-template <typename T, int ExtraFlags = array::forcecast> class array_t : public array {
+template <typename T, int ExtraFlags = array<>::forcecast> class array_t : public array<> {
```

Comment:
Why is `array_t` left out?  I'd think taking it as an extra argument then inheriting from `array<access_policy>` would make a lot of sense here.

This isn't related to your PR, but I also don't understand why we take ExtraFlags as a template argument here instead of just as an argument to `ensure()`.